pyquil/_parser/parser.py

```diff
@@ -0,0 +1,554 @@
+import pkgutil
+import operator
+from typing import List
+
+from lark import Lark, Transformer, v_args
+
+import numpy as np
+
+from pyquil.quilbase import (
+    AbstractInstruction,
+    DefGate,
+    DefPermutationGate,
+    DefGateByPaulis,
+    # TODO DefCircuit
```

Comment:
TODO

Reply:
This already exists as a TODO in the backlog so I've removed the comment